shopfloor/services/checkout.py

```diff
@@ -860,7 +859,16 @@ def _post_put_lines_in_package(self, lines_packaged):
     def _create_and_assign_new_packaging(self, picking, selected_lines, packaging=None):
         actions = self._actions_for("packaging")
         package = actions.create_package_from_packaging(packaging=packaging)
-        return self._put_lines_in_allowed_package(picking, selected_lines, package)
+        return self._pack_lines(picking, selected_lines, package)
+
+    def _pack_lines(self, picking, selected_lines, package):
+        lines_to_pack = selected_lines.filtered(self._filter_lines_to_pack)
+        response = self._put_lines_in_allowed_package(picking, lines_to_pack, package)
```

Comment:
* move check for `message = _("No line to pack found")` out of `_put_lines_in_allowed_package` and put it before this line.
* move `message = self.msg_store.goods_packed_in(package)` out of `_put_lines_in_allowed_package` and put it at the end of this method
* if `_put_lines_in_allowed_package` returns a response, return it immediately without posting (case coming from measurement) as we need to collect the measurement before posting the move.
* add `_auto_post_lines` in measurement module after the measurement have been successfully collected